pkg/csi_driver/controller.go

```diff
@@ -417,36 +437,99 @@ func (s *controllerServer) ControllerGetCapabilities(ctx context.Context, req *c
 	}, nil
 }
 
-// getRequestCapacity returns the volume size that should be provisioned
-func getRequestCapacity(capRange *csi.CapacityRange) (int64, error) {
-	if capRange == nil {
-		return minVolumeSize, nil
+// getTierFromParams returns the provided tier or default
+func getTierFromParams(params map[string]string) string {
+	if val, ok := params[paramTier]; ok {
+		return val
 	}
 
-	rCap := capRange.GetRequiredBytes()
-	rSet := rCap > 0
-	lCap := capRange.GetLimitBytes()
-	lSet := lCap > 0
+	return defaultTier
+}
+
+func invalidCapacityRange(capRange *csi.CapacityRange, tier string)  error {
+	validRange := provisionableCapacityForTier(tier)
+
+	minRequired := capRange.GetRequiredBytes()
```

Comment:
consider change this to `requiredCap` to align with naming in `CapacityRange`

Reply:
Done.